pg_search/src/postgres/utils.rs

```diff
@@ -86,52 +87,80 @@ pub struct CategorizedFieldData {
-        };
+            search_fields
+                .into_iter()
+                .map(|search_field| {
+                    let array_type = unsafe { pg_sys::get_element_type(attribute_type_oid) };
+                    let (base_oid, is_array) = if array_type != pg_sys::InvalidOid {
+                        (array_type, true)
+                    } else {
+                        (attribute_type_oid, false)
+                    };
 
-        for search_field in search_fields {
-            let array_type = unsafe { pg_sys::get_element_type(attribute_type_oid.value()) };
-            let (base_oid, is_array) = if array_type != pg_sys::InvalidOid {
-                (PgOid::from(array_type), true)
-            } else {
-                (attribute_type_oid, false)
-            };
+                    let base_oid = PgOid::from(base_oid);
+                    let is_json = matches!(
+                        base_oid,
+                        PgOid::BuiltIn(pg_sys::BuiltinOid::JSONBOID | pg_sys::BuiltinOid::JSONOID)
+                    );
+                    (
+                        search_field.clone(),
+                        CategorizedFieldData {
+                            attno,
+                            base_oid,
+                            is_array,
+                            is_json,
+                        },
+                    )
+                })
+                .collect::<Vec<_>>()
+        })
+        .collect()
+}
 
-            let is_json = matches!(
-                base_oid,
-                PgOid::BuiltIn(pg_sys::BuiltinOid::JSONBOID | pg_sys::BuiltinOid::JSONOID)
-            );
+/// Extracts the field attributes from the index relation.
+/// It returns a vector of tuples containing the field name and its type OID.
+pub fn extract_field_attributes(indexrel: &PgRelation) -> Vec<(String, Oid)> {
+    let tupdesc = unsafe { PgTupleDesc::from_pg_unchecked(indexrel.rd_att) };
+    let index_info = unsafe { pg_sys::BuildIndexInfo(indexrel.as_ptr()) };
+    let expressions = unsafe { PgList::<pg_sys::Expr>::from_pg((*index_info).ii_Expressions) };
+    let mut expressions_iter = expressions.iter_ptr();
 
-            categorized_fields.push((
-                search_field.clone(),
-                CategorizedFieldData {
-                    attno,
-                    base_oid,
-                    is_array,
-                    is_json,
-                },
-            ));
-        }
+    let mut field_attributes = Vec::new();
+    for i in 0..(unsafe { *index_info }).ii_NumIndexAttrs {
+        let heap_attno = (unsafe { *index_info }).ii_IndexAttrNumbers[i as usize];
+        let (attname, attribute_type_oid) = if heap_attno == 0 {
+            // Is an expression.
+            let Some(expression) = expressions_iter.next() else {
+                panic!("Expected expression for index attribute {i}.");
+            };
+            let node = expression.cast();
+            (format!("_pg_search_{}", i), unsafe {
```

Comment:
Please extract this constant prefix string from the three+ places it is used into a static somewhere ... probably in this file (if that doesn't cause cycles).